Tasks/Common/docker-common-v2/containerconnection.ts

```diff
@@ -23,7 +23,13 @@ export default class ContainerConnection {
     private oldDockerConfigContent: string;
 
     constructor() {
-        this.dockerPath = tl.which("docker", true);
+        if(process.env["RUNNING_ON"] == "KUBERNETES") {
+            this.dockerPath = tl.which("img", true);
```

Comment:
Is it 'Buildctl'?

Reply:
yes to push the image docker login is required so instead we will be using img login for images build using buildctl